File: sources/fsharp/07_string_append/07_string_append.fs
// task 07 string_append — expected output: 1000000
// build: dotnet build -c Release 07_string_append.fsproj    run: dotnet run -c Release (or bin/Release/net8.0/07_string_append.dll)

// System.String is immutable, so every + copies the whole string.
let mutable text = ""
let mutable i = 0
while i < 1000000 do
    text <- text + "x"
    i <- i + 1

printfn "%d" text.Length

File: sources/csharp/07_string_append/07_string_append.cs
// task 07 string_append — expected output: 1000000
// build: coreclr: dotnet build -c Release | nativeaot: dotnet publish -c Release -p:PublishAot=true | mono: mcs -optimize+ 07_string_append.cs    run: coreclr: dotnet run -c Release (or bin/Release/net8.0/07_string_append.exe) | nativeaot: bin/Release/net8.0/publish/07_string_append.exe | mono: mono 07_string_append.exe
using System;

class Program
{
    static void Main()
    {
        // Plain string concatenation: System.String is immutable, so this copies every time.
        string text = "";
        for (int i = 0; i < 1000000; i++)
        {
            text = text + "x";
        }
        Console.WriteLine(text.Length);
    }
}
